plugins/cfu/README.md

```diff
@@ -0,0 +1,35 @@
+# Component Firmware update
+
+## Introduction
+
+CFU is a protocol from Microsoft to make it easy to install firmware on HID devices.
+
+This protocol is unique in that it requires has a pre-download phase before sending the firmware to
+the microcontroller. This is so the device can check if the firmware is required and compatible.
+CFU also requires devices to be able to transfer the entire new transfer mode in runtime mode.
+
+This plugin supports the following protocol ID:
+
+* com.microsoft.cfu
+
+## GUID Generation
+
+These devices use standard USB DeviceInstanceId values, as well as two extra for the component ID
+and the bank, e.g.
+
+* `USB\VID_17EF&PID_7226&CID_01&BANK_1`
+* `USB\VID_17EF&PID_7226&CID_01`
+* `USB\VID_17EF&PID_7226`
+
+## Update Behavior
+
+The device has to support runtime updates and does not have a detach-into-bootloader mode -- but
+after the install has completed the device still has to reboot into the new firmware.
+
+## Vendor ID Security
+
+The vendor ID is set from the USB vendor, in this instance set to `USB:0x17EF`
```

Comment:
I don't see why CFU has to be USB

Reply:
Indeed, valid point -- and the same with the Instance IDs too. I guess well get HIDRAW for both thanks to the `FuUdevDevice->probe()` -- I've changed the docs and code accordingly.